### PR TITLE
telemetry: change LogVerbosity to an enum, use it throughout public API

### LINKED ISSUE #13: For API consistency make `LogVerbosity` an enum and use it across the API

### PR DESCRIPTION
**Pull request by evanrittenhouse**
Closes #13.

### CODE REVIEW

File: foundations/src/telemetry/log/init.rs [hunk 1]
```diff
@@ -137,7 +137,7 @@ where
         drain,
         FieldRedactFilterFactory::new(settings.redact_keys.clone()),
     );
-    let drain = drain.filter_level(*settings.verbosity);
+    let drain = drain.filter_level((&settings.verbosity).into());
```
Comment (inikulin) [inline foundations/src/telemetry/log/init.rs:140]: We needed deref before as we were wrapping `slog::Level`. Considering that now we have `into` and verbosity is a `Copy` type, we can just have `settings.verbosity.into()` here

**Review by inikulin — CHANGES_REQUESTED** [review]

File: foundations/src/telemetry/settings/logging.rs [hunk 2]
```diff
@@ -66,21 +65,58 @@ pub enum LogFormat {
     Json,
 }
 
-/// Verbosity level of the log.
+/// Log verbosity levels which match 1:1 with [`slog::Level`].
 #[derive(Clone, Debug, Copy)]
-pub struct LogVerbosity(pub Level);
+pub enum LogVerbosity {
```
Comment (inikulin) [inline foundations/src/telemetry/settings/logging.rs:70]: This needs settings attrs as for other types above

File: foundations/src/telemetry/settings/logging.rs [hunk 3]
```diff
@@ -66,21 +65,58 @@ pub enum LogFormat {
...
+    /// See [`slog::Level::Debug`].
+    Debug,
+    /// See [`slog::Level::Trace`].
+    Trace,
+}
 
 impl Default for LogVerbosity {
     fn default() -> Self {
-        Self(Level::Info)
+        Self::Info
```
Comment (inikulin) [inline foundations/src/telemetry/settings/logging.rs:87]: nit: you can use derive `Default` in combination with `#[default]` attribute for the variant instead

**Review by inikulin — CHANGES_REQUESTED** [review]

File: foundations/src/telemetry/settings/logging.rs (hunk 2 above)
Comment (evanrittenhouse) [reply foundations/src/telemetry/settings/logging.rs:70]: Hmm, isn't there an `impl Settings` already? https://github.com/cloudflare/foundations/pull/14/files#diff-9eae3d0e9cd889fe453ebfadef152fa2ab92eae3cbb086ac9e32c7dbd46b757bR158

Comment (inikulin) [reply foundations/src/telemetry/settings/logging.rs:70]: Right, was hidden by the diff for me. But now, when we have an enum and not a wrapper, we can do it via macro

Comment (evanrittenhouse) [reply foundations/src/telemetry/settings/logging.rs:69]: Got it, done

File: foundations/src/telemetry/settings/logging.rs [hunk 4]
```diff
@@ -66,21 +63,52 @@ pub enum LogFormat {
...
-pub struct LogVerbosity(pub Level);
+/// Log verbosity levels which match 1:1 with [`slog::Level`].
+#[cfg_attr(
+    feature = "settings",
+    settings(crate_path = "crate", impl_default = false)
+)]
+#[cfg_attr(not(feature = "settings"), derive(Clone, Debug))]
+#[derive(Copy, Default)]
+pub enum LogVerbosity {
+    /// See [`slog::Level::Critical`].
```
[the remark this answers is not in the document]
Comment (evanrittenhouse) [reply foundations/src/telemetry/settings/logging.rs:74]: Unfortunately the `slog` docs aren't very descriptive: https://docs.rs/slog/latest/slog/enum.Level.html#variant.Critical.

Maybe we just say "Matches `slog::Level::Critical`?

File: foundations/src/telemetry/settings/logging.rs [hunk 5]
```diff
@@ -66,21 +63,52 @@ pub enum LogFormat {
     Json,
 }
 
-/// Verbosity level of the log.
-#[derive(Clone, Debug, Copy)]
-pub struct LogVerbosity(pub Level);
+/// Log verbosity levels which match 1:1 with [`slog::Level`].
```
[the remark this answers is not in the document]
Comment (evanrittenhouse) [reply foundations/src/telemetry/settings/logging.rs:66]: We may want to remove this note entirely depending on what we decide to do with my comment above

**Pull request merged**